custom/basic-call/components/Call/VideoGrid.js

```diff
@@ -45,65 +47,61 @@ export const VideoGrid = React.memo(
       };
     }, []);
 
-    // Basic brute-force packing algo
-    const layout = useMemo(() => {
-      const aspectRatio = DEFAULT_ASPECT_RATIO;
-      const tileCount = participants.length || 0;
-      const w = dimensions.width;
-      const h = dimensions.height;
-
-      // brute-force search layout where video occupy the largest area of the container
-      let bestLayout = {
-        area: 0,
-        cols: 0,
-        rows: 0,
-        width: 0,
-        height: 0,
-      };
-
-      for (let cols = 0; cols <= tileCount; cols += 1) {
-        const rows = Math.ceil(tileCount / cols);
-        const hScale = w / (cols * aspectRatio);
-        const vScale = h / rows;
-        let width;
-        let height;
-        if (hScale <= vScale) {
-          width = Math.floor(w / cols);
-          height = Math.floor(width / aspectRatio);
+    const [tileWidth, tileHeight] = useMemo(() => {
+      const width = Math.floor(dimensions.width);
+      const height = Math.floor(dimensions.height);
+      const tileCount = participantCount || 0;
+      if (tileCount === 0) return [width, height];
+      const dims = [];
+      /**
+       * We'll calculate the possible tile dimensions for 1 to n columns.
+       */
+      for (let columnCount = 1; columnCount <= tileCount; columnCount++) {
+        // Pixels used for flex gap between tiles
+        const columnGap = columnCount - 1;
+        let maxWidthPerTile = Math.floor((width - columnGap) / columnCount);
+        let maxHeightPerTile = Math.floor(
+          maxWidthPerTile / DEFAULT_ASPECT_RATIO
+        );
+        const rowCount = Math.ceil(tileCount / columnCount);
+        const rowGap = rowCount - 1;
+        if (rowCount * maxHeightPerTile + rowGap > height) {
```

Comment:
Also just confirming: are we expecting to only be adding one count of `rowGap` here? Or is the rowGap intended to be per tile? (ie should this be `rowCount * (maxHeightPerTile + rowGap)`?